src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -259,8 +260,17 @@ namespace cryptonote
 
     m_fakechain = test_options != NULL;
     bool r = handle_command_line(vm);
+    bool testnet = command_line::get_arg(vm, command_line::arg_testnet_on);
+    auto p2p_bind_arg = testnet ? nodetool::arg_testnet_p2p_bind_port : nodetool::arg_p2p_bind_port;
+    std::string m_port = command_line::get_arg(vm, p2p_bind_arg);
+    std::string m_config_folder_mempool = m_config_folder;
+
+    if ((!testnet && m_port != std::to_string(::config::P2P_DEFAULT_PORT))
+        || (testnet && m_port != std::to_string(::config::testnet::P2P_DEFAULT_PORT))) {
+      m_config_folder_mempool = m_config_folder_mempool + "/" + m_port;
```

Comment:
Can this break on Windows ? The rest of the code uses boost::path, which has a / operator for this.